### DISCUSSION TITLE
RV-C over CAN

### DISCUSSION THREAD

**Discussion by AllonisDavid** [spbrogan/rvc2mqtt#123] [General]
Hello Sean,
I wonder if I could ask you a few questions about your RVC2MQTT program. Very nice work. I have it running on a RPI4 with RS485 CAN Hat and connected to two RV-C devices, a TANK SENSOR and HVAC controller.

The problem is I don't ever see anything at all on the CAN bus from the RV-C devices.

I should add that I have a second RPI4 and CAN hat on the same bus monitoring CAN traffic and I do see request messages sent on CAN from RVC2MQTT without fail, just no responses from RV-C. Sometimes I get back an error frame after a request, but not always.

I can also send mock RV-C messages from the second RPI4 and see them land in RVC2MQTT and be decoded, resulting in a MQTT message,  so I am sure the CAN network and RVC2MQTT program are working properly.

So my questions is, do you have any clues as to what could be wrong with our RV-C devices causing them to not talk?

My understand is that both of the RV-C devices perodically transmit a status message that we should see just by listening.

Thanks
Overworked Developer

**Comment by spbrogan** [comment]
The RVC2MQTT sw has been developed to connect onto a functioning/active RV-C network and then sniff the activity and talk to the devices.  I wonder if there is more to setup the network and get devices talking.  In the RV-C spec there is discussion about how devices claim addresses (section 3.3).  Also, in section 4 it talks about required messages, you might try sending out some of those and see if any device responds.  

Can you share more details about your setup?  What Tank Sensor or HVAC controller are you using?

**Reply by AllonisDavid** [reply]
We have the CAN network setup for testing.  120ohm resisters on both ends of the High / Low CAN bus as terminators.  In the middle, we have two Pis, one running a Waveshare RS485/ CAN hat, and the other a CopperHillTech PiCan2 https://copperhilltech.com/pican2-can-interface-for-raspberry-pi-with-smps/.  

We have a SeeLevel 709RVC fluid and propane level sensor.  Additionally we have a Dometic HVAC RV-C interface Dometic Climate Control Gateway RV-C Canbus Network Interface 20 FT Kit -
3316945.021  

We have a CAN logger running and don't see any CAN traffic when either of the two RV-C devices are powered up.

Thanx for any insights!

**Comment by spbrogan** [comment]
the waveshare and most "can hats" have resisters built in (and a jumper which is usually on by default).  Any chance you are "double" terminating the bus?

**Reply by AllonisDavid** [reply]
hmmmm, something to look at.  I am able to send CAN traffic between the two Pis.

**Reply by AllonisDavid** [reply]
Pulled both resistors off and put one Pi on one end and the second on the other...still no RV-C data yet the Pis can talk CAN to each other.  Weird....

**Comment by spbrogan** [comment]
have you hacked up the pi to send the address identification or status commands?

**Comment by AllonisDavid** [comment]
Ends up we were testing with a defective SeeLevel RV-C display.  Garett sent a replacement out and immediate on wiring, RV-C messages started flowing!  We now have RV-C data being sent to myServer 6 via MQTT!  Thanx.
